File: app/src/main/java/com/limelight/binding/input/GameInputDevice.java
package com.limelight.binding.input;

import com.limelight.GameMenu;

import java.util.List;

/**
 * Description
 * Date: 2024-01-16
 * Time: 15:26
 * User: Genng(genng1991@gmail.com)
 */
public interface GameInputDevice {

    /**
     * @return list of device specific game menu options, e.g. configure a controller's mouse mode
     */
    List<GameMenu.MenuOption> getGameMenuOptions();
}

File: app/src/main/java/com/limelight/binding/input/GameInputDevice.kt
package com.limelight.binding.input

import com.limelight.gamemenu.GameMenu

/**
 * Generic Input Device
 */
interface GameInputDevice {
    /**
     * @return device-specific actions that should remain immediately reachable in the game menu
     */
    fun getGameMenuQuickOptions(): List<GameMenu.MenuOption>

    /** Called after a game menu opened for this input device is dismissed. */
    fun onGameMenuDismissed() {}
}
